### ISSUE TITLE
Crash on F25 w/ 1.0.1-5

### ISSUE THREAD

**Issue by xenithorb** [manatools/dnfdragora#44]
Package version: 
```
dnfdragora-1.0.1-5.git20170505.2a3b056.fc25.noarch
dnfdragora-gui-1.0.1-5.git20170505.2a3b056.fc25.noarch
```

Error:

An orange dialog window shows and presents the following:
```
dnfdaemon client error occurred:
g-io-error-quark: GDBus.Error:org.baseurl.DnfSystem.LockedError: dnf is locked by another application (36)
```

Full console log:

```
Skipped exception: <[Errno 2] No such file or directory: './dnfdragora.yaml'> 
<_M_> [ui] YUILoader.cc:50 loadUI(): DISPLAY: ":0"
<_M_> [ui] YUILoader.cc:51 loadUI(): XDG_CURRENT_DESKTOP: "KDE"
<_M_> [ui] YUILoader.cc:52 loadUI(): YUI_PREFERED_BACKEND: ""
<_M_> [ui] YUILoader.cc:86 loadUI(): User-selected UI-plugin: ""
<_M_> [ui] YPath.cc:49 YPath(): Given filename: libyui-gtk.so.8
<_M_> [ui] YSettings.cc:74 progDir(): progDir: ""
<_M_> [ui] YPath.cc:65 YPath(): Preferring subdir: 
<_M_> [ui] YPath.cc:66 YPath(): Subdir given with filename: 
<_M_> [ui] YPath.cc:67 YPath(): Looking for: libyui-gtk.so.8
<_M_> [ui] YPath.cc:138 lsDir(): Looking in /usr/lib64/yui
<_M_> [ui] YPath.cc:118 YPath(): Found libyui-gtk.so.8 in /usr/lib64/yui
<_M_> [ui] YPath.cc:49 YPath(): Given filename: libyui-ncurses.so.8
<_M_> [ui] YSettings.cc:74 progDir(): progDir: ""
<_M_> [ui] YPath.cc:65 YPath(): Preferring subdir: 
<_M_> [ui] YPath.cc:66 YPath(): Subdir given with filename: 
<_M_> [ui] YPath.cc:67 YPath(): Looking for: libyui-ncurses.so.8
<_M_> [ui] YPath.cc:138 lsDir(): Looking in /usr/lib64/yui
<_M_> [ui] YPath.cc:118 YPath(): Found libyui-ncurses.so.8 in /usr/lib64/yui
<_M_> [ui] YPath.cc:49 YPath(): Given filename: libyui-qt.so.8
<_M_> [ui] YSettings.cc:74 progDir(): progDir: ""
<_M_> [ui] YPath.cc:65 YPath(): Preferring subdir: 
<_M_> [ui] YPath.cc:66 YPath(): Subdir given with filename: 
<_M_> [ui] YPath.cc:67 YPath(): Looking for: libyui-qt.so.8
<_M_> [ui] YPath.cc:138 lsDir(): Looking in /usr/lib64/yui
<_M_> [ui] YPath.cc:118 YPath(): Found libyui-qt.so.8 in /usr/lib64/yui
<_M_> [ui] YUILoader.cc:118 loadUI(): Using UI-plugin: "qt"
<_M_> [ui] YSettings.cc:179 loadedUI(): Set loadedUI to "qt"
<_M_> [ui] YSettings.cc:180 loadedUI(): loadedUI is now locked.
<_M_> [ui] YPath.cc:49 YPath(): Given filename: libyui-qt.so.8
<_M_> [ui] YSettings.cc:74 progDir(): progDir: ""
<_M_> [ui] YPath.cc:65 YPath(): Preferring subdir: 
<_M_> [ui] YPath.cc:66 YPath(): Subdir given with filename: 
<_M_> [ui] YPath.cc:67 YPath(): Looking for: libyui-qt.so.8
<_M_> [ui] YPath.cc:138 lsDir(): Looking in /usr/lib64/yui
<_M_> [ui] YPath.cc:118 YPath(): Found libyui-qt.so.8 in /usr/lib64/yui
<_M_> [ui] YUI.cc:76 YUI(): This is libyui 3.3.2
<_M_> [ui] YUI.cc:77 YUI(): Creating UI without threads
<_M_> [qt-ui] YQUI.cc:100 YQUI(): This is libyui-qt 2.47.1
<_M_> [ui] YUI.cc:230 topmostConstructorHasFinished(): Running without threads
<_M_> [qt-styler] QY2Styler.cc:141 loadStyleSheet(): Couldn't open style sheet "/usr/share/libyui/theme/current/wizard/style.qss"
<_M_> [qt-ui] YQUI.cc:296 processCommandLineArgs(): Qt argument: /usr/bin/python3
<_M_> [qt-ui] YQUI.cc:296 processCommandLineArgs(): Qt argument: /usr/bin/dnfdragora
<_M_> [qt-ui] YQUI.cc:439 calcDefaultSize(): Default size: 1344 x 723
<ERR> [qt-ui] YQUI.cc:262 initUI(): Forcing /libQtGui.so.5 open failed
<_M_> [qt-ui] YQApplication.cc:133 loadPredefinedQtTranslations(): Selected language: en_US
<WRN> [qt-ui] YQApplication.cc:137 loadPredefinedQtTranslations(): Qt locale directory not set - no translations for predefined Qt dialogs
<_M_> [qt-ui] YQUI.cc:273 initUI(): YQUI initialized. Thread ID: 0x0x7f3ec1db6700
<_M_> [ui] YSettings.cc:197 loadedUI(): loadedUI: "qt"
<_M_> [ui] YPath.cc:49 YPath(): Given filename: libyui-mga-qt.so.8
<_M_> [ui] YSettings.cc:74 progDir(): progDir: ""
<_M_> [ui] YPath.cc:65 YPath(): Preferring subdir:
<_M_> [ui] YPath.cc:66 YPath(): Subdir given with filename:
<_M_> [ui] YPath.cc:67 YPath(): Looking for: libyui-mga-qt.so.8
<_M_> [ui] YPath.cc:138 lsDir(): Looking in /usr/lib64/yui
<_M_> [ui] YPath.cc:118 YPath(): Found libyui-mga-qt.so.8 in /usr/lib64/yui
<_M_> [ui] YPath.cc:49 YPath(): Given filename: libyui-mga-qt.so.8
<_M_> [ui] YSettings.cc:74 progDir(): progDir: ""
<_M_> [ui] YPath.cc:65 YPath(): Preferring subdir:
<_M_> [ui] YPath.cc:66 YPath(): Subdir given with filename:
<_M_> [ui] YPath.cc:67 YPath(): Looking for: libyui-mga-qt.so.8
<_M_> [ui] YPath.cc:138 lsDir(): Looking in /usr/lib64/yui
<_M_> [ui] YPath.cc:118 YPath(): Found libyui-mga-qt.so.8 in /usr/lib64/yui
<_M_> [ew] YExternalWidgets.cc:37 YExternalWidgets(): Creating Libyui External Widgets object
<_M_> [mga-qt-ui] YMGA_QCBTable.cc:74 YMGA_QCBTable():  Slection mode 2
dnfdaemon client error: g-io-error-quark: GDBus.Error:org.baseurl.DnfSystem.LockedError: dnf is locked by another application (36)
<_M_> [ui] YUILoader.cc:163 deleteUI(): Shutting down UI
Closing dnfdragora
<_M_> [ew] YExternalWidgets.cc:116 ~YExternalWidgetsTerminator(): Shutting down mga External Widgets
<_M_> [ew] YMGAWidgetExtensionFactory.cc:49 ~YMGAWidgetFactory(): Factory removed
```

Temporary resolution:

```
systemctl stop dnfdaemon
```
Then start `dnfdragora` again, and it works ok. 

Only error associated with dnfdaemon log (not sure if related or not)

```
May 07 07:54:10 icarus.net.overtmind.com dnfdaemon-system[23070]: The 'group_persistor' function is not a part of DNF API and will be removed in the upcoming DNF release. Please use only officially supported API functions. DNF API documentation is available at https://dnf.readthedocs.org/en/latest/api.html.
```

**Comment by anaselli** [comment]
Sorry which is the problem? I mean wasn't dnf db really locked? or it was due to a dnfdragora crash? and you could not run it any more? 
From what i see here
> <_M_> [mga-qt-ui] YMGA_QCBTable.cc:74 YMGA_QCBTable():  Slection mode 2
> dnfdaemon client error: g-io-error-quark: GDBus.Error:org.baseurl.DnfSystem.LockedError: dnf is locked by another application (36)
> <_M_> [ui] YUILoader.cc:163 deleteUI(): Shutting down UI
> Closing dnfdragora
> <_M_> [ew] YExternalWidgets.cc:116 ~YExternalWidgetsTerminator(): Shutting down mga External Widgets
> <_M_> [ew] YMGAWidgetExtensionFactory.cc:49 ~YMGAWidgetFactory(): Factory removed
dnfdragora exited correctly for that.

**Comment by xenithorb** [comment]
DNF was still able to be used, it wasn't locked. I don't know the inner workings of `dnfdaemon` but that seemed to be the culprit. But all dnfdragora did was present some weird orange looking window and then quit immediately. I thought that was strange so I reported it here.

**Comment by anaselli** [comment]
Well the reason why it was locked could be. For instance as in yumex-dnf if for any reasons crashes and does not unlock the db then it cannot start again. So they both offer a "--exit" option to force db unlocking. But if any other applications such as the updater for instance had locked db to look for updates, then that is normal.

**Comment by xenithorb** [comment]
I purposefully don't have anything like PackageKit going, so the only other thing it could have been would be dnf itself and that process had exited cleanly some minutes beforehand. `dnfdaemon` was stuck in an incorrect state, I think?

**Comment by jcubic** [comment]
i had the same problem even that I could run dnf from terminal I executed dnfdragora --exit and it solve the issue

**Comment by jcubic** [comment]
I don't remember it just keeping showing error. I'm using Fedora 27 XFce

**Comment by anaselli** [comment]
I think this bug should be fixed with 2.0.x version, since the way how dnfdragora locks and use db is changed, moreover dnfdragora-update just tries to lock it and if not possible it retries later to check for updates.
